test/regress/expect/tpch0001_d/q03.txt

```diff
@@ -21,35 +21,35 @@ order by
 	revenue desc,
 	o_orderdate
 limit 10
-Total cost: 23009.8, memory=267728
-PhysicLimit (10) (inccost=23009.8, cost=10, rows=10) (actual rows=8)
+Total cost: 28022.9, memory=62600
+PhysicLimit (10) (inccost=28022.9, cost=10, rows=10) (actual rows=8)
     Output: l_orderkey[0],{sum(l_extendedprice*1-l_discount)}[1],o_orderdate[2],o_shippriority[3]
-    -> PhysicOrder  (inccost=22999.8, cost=2851.9, rows=458, memory=10992) (actual rows=8)
+    -> PhysicOrder  (inccost=28012.9, cost=2851.9, rows=458, memory=10992) (actual rows=8)
         Output: l_orderkey[0],{sum(l_extendedprice*1-l_discount)}[1],o_orderdate[2],o_shippriority[3]
         Order by: {sum(l_extendedprice*1-l_discount)}[1], o_orderdate[2]
-        -> PhysicHashAgg  (inccost=20147.89, cost=1374, rows=458, memory=21984) (actual rows=8)
+        -> PhysicHashAgg  (inccost=25161, cost=1374, rows=458, memory=21984) (actual rows=8)
             Output: {l_orderkey}[0],{sum(l_extendedprice*1-l_discount)}[3],{o_orderdate}[1],{o_shippriority}[2]
             Aggregates: sum(l_extendedprice[4]*1-l_discount[7])
             Group by: l_orderkey[0], o_orderdate[1], o_shippriority[2]
-            -> PhysicGather 1 : 10 (inccost=18773.89, cost=45.8, rows=458) (actual rows=14)
+            -> PhysicGather 1 : 10 (inccost=23787, cost=4580, rows=458) (actual rows=14)
                 Output: l_orderkey[2],o_orderdate[3],o_shippriority[4],{l_extendedprice*1-l_discount}[5],l_extendedprice[6],{1-l_discount}[7],{1}[0],l_discount[8]
-                -> PhysicHashJoin  (inccost=18728.09, cost=2097, rows=458, memory=464) (actual rows=0)
+                -> PhysicHashJoin  (inccost=19207, cost=2097, rows=458, memory=464) (actual rows=0)
                     Output: l_orderkey[2],o_orderdate[3],o_shippriority[4],{l_extendedprice*1-l_discount}[5],l_extendedprice[6],{1-l_discount}[7],{1}[0],l_discount[8]
                     Filter: c_custkey[1]=o_custkey[9]
                     -> PhysicScanTable customer (inccost=150, cost=150, rows=29) (actual rows=0)
```

Comment:
here we choose to redistribute join 1581 rows instead of broadcast/redistribute customer as build side?

Reply:
I didn't have redistribution enforcement yet, I will add that